feat: add fortune telling app (#62167)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/blocks/workshop-fortune-teller-app/68ed4585f6a1d6e07b9d09c5.md

---
id: 68ed4585f6a1d6e07b9d09c5
title: Step 23
challengeType: 0
dashedName: step-23
---

# --description--

Inside the fetchCardsData method, declare variable named response, assign to by calling the fetch function with the following URL:`${CDN_URL}/card_data.json`.

> Add the await keyword before the fetch call, and wrap the entire logic inside a try...catch block.

Reviewer: ```suggestion
Add the `await` keyword before the `fetch` call, and wrap the entire logic inside a `try...catch` block.
```

Make sure to include error as a parameter in the catch block so you can log or handle it properly.

# --hints--

> You should have declared `response` variable inside the fetchCardsData method.

Reviewer: ```suggestion
You should declare a `response` variable inside the `fetchCardsData` method.
```

```js
assert.match(code, /fetchCardsData\s*\([^)]*\)\s*{[\s\S]*?\bresponse\b/);
```

> Your response variable should have assigned to `fetch` function with `${CDN_URL}/card_data.json` url.

Reviewer: ```suggestion
Your `response` variable should be assigned to the `fetch` function with the `${CDN_URL}/card_data.json` URL.
```

> 

Reviewer: ```suggestion
```

```js
assert.match(code, /const\s+response\s*=\s*await\s+fetch\s*\(\s*`\$\{CDN_URL\}\/card_data\.json`\s*\)/
);
```

Your fetch function should have `await` keyword.

```js
assert.match(code, /response\s*=\s*await\s+fetch\s*\(/);
```

> You should have wrapped your fetch function with `try-catch` block.

Reviewer: ```suggestion
You should wrap your `fetch` function with a `try-catch` block.
```

```js
assert.match(code, /try\s*{[\s\S]*?await\s+fetch[\s\S]*?}\s*catch\s*\(/);
```

Your catch block should have error parameter.

```js
assert.match(code, /catch\s*\(\s*error\s*\)/);
```

# --seed--

## --seed-contents--

```html

> <!doctype html>

Reviewer: ```suggestion
<!DOCTYPE html>
```
